pkg/common/utils/resource/service.go

```diff
@@ -254,27 +255,40 @@ func getMetaServiceContainerPorts(config map[string]interface{}) []corev1.Contai
 }
 
 func getFeContainerPorts(config map[string]interface{}) []corev1.ContainerPort {
-	return []corev1.ContainerPort{{
-		Name:          GetPortKey(HTTP_PORT),
-		ContainerPort: GetPort(config, HTTP_PORT),
-		Protocol:      corev1.ProtocolTCP,
-	}, {
-		Name:          GetPortKey(RPC_PORT),
-		ContainerPort: GetPort(config, RPC_PORT),
-		Protocol:      corev1.ProtocolTCP,
-	}, {
-		Name:          GetPortKey(QUERY_PORT),
-		ContainerPort: GetPort(config, QUERY_PORT),
-		Protocol:      corev1.ProtocolTCP,
-	}, {
-		Name:          GetPortKey(EDIT_LOG_PORT),
-		ContainerPort: GetPort(config, EDIT_LOG_PORT),
-		Protocol:      corev1.ProtocolTCP,
-	}}
-}
+	ports := []corev1.ContainerPort{
+		{
+			Name:          GetPortKey(HTTP_PORT),
+			ContainerPort: GetPort(config, HTTP_PORT),
+			Protocol:      corev1.ProtocolTCP,
+		}, {
+			Name:          GetPortKey(RPC_PORT),
+			ContainerPort: GetPort(config, RPC_PORT),
+			Protocol:      corev1.ProtocolTCP,
+		}, {
+			Name:          GetPortKey(QUERY_PORT),
+			ContainerPort: GetPort(config, QUERY_PORT),
+			Protocol:      corev1.ProtocolTCP,
+		}, {
+			Name:          GetPortKey(EDIT_LOG_PORT),
+			ContainerPort: GetPort(config, EDIT_LOG_PORT),
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
 
+	arrowFlightPort := GetPort(config, ARROW_FLIGHT_SQL_PORT)
+	if arrowFlightPort != -1 {
+		ports = append(ports, corev1.ContainerPort{
+			// GetPortKey(ARROW_FLIGHT_SQL_PORT) Hardcoding for now. In Kubernetes, container port names must be no more than 15 characters.
```

Comment:
You can modify `func GetPortKey()` so that `GetPortKey(ARROW_FLIGHT_SQL_PORT)` returns `"arrow-flight"`, and then change it here to `GetPortKey(ARROW_FLIGHT_SQL_PORT)`